### PR TITLE
feat: add tfsec (v1.28.1) on all platforms

### PR DESCRIPTION
**Pull request by Kevin-CB**
For https://github.com/jenkins-infra/packer-images/issues/198

This PR add tfsec on all platforms

### CODE REVIEW

**Comment by lemeurherve** [comment]
Thanks for this new PR @Kevin-CB 👏 

FYI there are two issues but I haven't found yet why they're occuring:
- the sanity check is apparently running `tfsec-checkgen` instead of `tfsec`, and this tool doesn't know the `--version` flag, thus failing the (Linux) builds
- This sanity check returning the same error message on Windows surprisingly doesn't fail the corresponding Windows builds

I'll take another look later.

File: provisioning/ubuntu-provision.sh [hunk 1]
```diff
@@ -532,6 +532,14 @@ function install_goss() {
   chmod +rx /usr/local/bin/goss
 }
 
+function install_tfsec() {
+  apt-get update --quiet
+  apt-get install --yes --no-install-recommends curl # Should already be there but this function should be autonomous
+
+  curl --silent --location --show-error "https://github.com/aquasecurity/tfsec/releases/download/v${TFSEC_VERSION}/tfsec-checkgen-linux-${ARCHITECTURE}" --output /usr/local/bin/tfsec
```
Comment (lemeurherve) [inline provisioning/ubuntu-provision.sh:539]:
```suggestion
  curl --silent --location --show-error "https://github.com/aquasecurity/tfsec/releases/download/v${TFSEC_VERSION}/tfsec-linux-${ARCHITECTURE}" --output /usr/local/bin/tfsec
```

I think the fact only tfsec-checkgen release assets are displayed by default induced you in error:
image

File: provisioning/windows-provision.ps1 [hunk 2]
```diff
@@ -296,6 +296,13 @@ $downloads = [ordered]@{
             & kubectl.exe version --client;
         }
     };
+    'tfsec' = @{
+        'url' = 'https://github.com/aquasecurity/tfsec/releases/download/v{0}/tfsec-checkgen-windows-amd64.exe' -f $env:TFSEC_VERSION;
```
Comment (lemeurherve) [inline provisioning/windows-provision.ps1:300]:
```suggestion
        'url' = 'https://github.com/aquasecurity/tfsec/releases/download/v{0}/tfsec-windows-amd64.exe' -f $env:TFSEC_VERSION;

```

**Review by lemeurherve — COMMENTED** [review]
Found the culprit for the first issue.

File: provisioning/ubuntu-provision.sh (hunk 1 above)
Comment (Kevin-CB) [reply provisioning/ubuntu-provision.sh]: Yeah indeed, you're right 😅 

Sorry about that and thanks for the fix!

**Comment by dduportal** [comment]
>     * This sanity check returning the same error message on Windows surprisingly doesn't fail the corresponding Windows builds (FYI @dduportal 👀)

As far as I can tell (ref. https://github.com/jenkins-infra/packer-images/pull/459/files#diff-40afc0140eabc567a7fffe74f8e456d714ec88e36e9aca911d2e95cd8696a40bR2), the powershell script fails when its instructions fails. Just tested it and it works as expected.

I wonder if the mechanism we use for the sanity check is not the culprit: my powershell-fu is not really good but from https://ss64.com/ps/call.html it looks to me that the exit code is not captured.

https://github.com/jenkins-infra/packer-images/issues/337 is the same (I'm adding more note on the issue to avoid polluting the discussion here)

**Review by lemeurherve — APPROVED** [review]

**Review by dduportal — APPROVED** [review]
Nice job, thanks a lot! 👏

**Pull request merged**